### PR TITLE
simd-doc: Add failing test for parsing incomplete rows

### PR DESCRIPTION
**Pull request by jshearer**
**Description:**

(Describe the high level scope of new or changed features)

**Workflow steps:**

(How does one use this feature, and how has it changed)

**Documentation links affected:**

(list any [documentation links](https://docs.google.com/document/d/1SRC9VS9zyCzWl3n4HXHbc4wPB1eLxJHkA2rtu9ZNokM/edit?usp=sharing) that you created, or existing ones that you've identified as needing updates, along with a brief description)

**Notes for reviewers:**

(anything that might help someone review this PR)

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/1734)

### CODE REVIEW

File: crates/simd-doc/src/tests/fixtures.rs [hunk 1]
```diff
@@ -166,76 +167,91 @@ fn test_incomplete_row_parsing() {
...
+                    },
+                ),
+            ],
+        ),
+        (
+            "\"test\": {\"this\": \"is\", \"a\": \"doc\"}}\n{\"this\": \"is a real doc\"}\n",
+            [
+                (
+                    -1,
+                    String("trailing characters at line 1 column 7 @ 0..36"),
```
Comment (jshearer) [inline crates/simd-doc/src/tests/fixtures.rs:197]: Where does this message come from? It doesn't match the new errors you added to the parser: `missing trailing newline`

**Comment by jshearer** [comment]
This LGTM btw -- I created the PR so I can't approve it, but would be good to merge this at some point, my question above isn't blocking (though I am curious)

**Review by jgraettinger — APPROVED** [review]
LGTM

File: crates/simd-doc/src/tests/fixtures.rs (hunk 1 above)
Comment (jgraettinger) [reply crates/simd-doc/src/tests/fixtures.rs:197]: Yep. The Parser attempts to use simd-json first, which almost always works. If simd-json fails, though, we throw a c++ exception which gets turned into a Result by the `cxx` crate.

We handle this by using a fallback path which does the same job but using serde_json, which produces better errors and is able to handle documents of larger sizes (simd-json is constrained to 1MB documents).

So tl;dr these errors are produced by the serde_json fallback after the "missing excepted newline" exception is thrown / caught / handled.

**Pull request merged**